### PR TITLE
Adds controller system ssh key to controller config.

### PR DESCRIPTION
**Pull request by tlm**
Every Juju controller has generated for it a system ssh key that allows the controller itself to ssh into machines and containers. Predominantly this has been used for configuring machine based firewalls.

A fix was introduced as part of #16394 that took the juju-system-ssh key as part of the controllers model config and appended it to newly created model's authorised key.

The problems this introduces is that it special cases the controller model as a defaults source for models which is not the design Juju is attempting to achieve. It also means that during model migrations we are migrating the system ssh key of one controller to another potentially introducing a security bug into the model where the previous controller could still access machines of a migrated model.

With this PR we have made the bootstrap client instead generate the system ssh key and place it into the controller's config. In the API facades we now form a union of the two ssh key sources before returning them to the caller.

This work also allows for the continuation of moving model config over to Dqlite as now we don't have to special case the controllers model config in the design.

As part of this PR we have also updated some wording around model defaults at bootstrap time that was confusing and not correct.

Newly created instance configs will also get the controllers system ssh key.

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing
- [x] [Integration tests](https://github.com/juju/juju/tree/main/tests), with comments saying what you're testing
- [x] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages

## QA steps

Reproduce the QA steps from this PR https://github.com/juju/juju/pull/10288:

```
juju bootstrap --config compartment-id=ocid1.compartment.oc1..aaaaaaaanvu2racnlczevenu73dlcf3nokgsjpdkbdgp4xbrz3lb2y4giyjq oci-canonical c
juju deploy ubuntu -n 2
juju deploy 'juju-qa-network-health'
juju expose network-health
juju add-relation ubuntu network-health
juju exec --unit ubuntu/0 curl <public IP of machine 1>:8039
```

The last command should `pass`:

```
juju run --unit ubuntu/0 curl <public IP of machine 1>:8039
pass  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100     4    0     4    0     0   2000      0 --:--:-- --:--:-- --:--:--  2000
```

also, if you ssh onto any created machine, you should see the correct authorized keys:

```
juju ssh 0
ubuntu@juju-6ddb09-0:~/.ssh$ cat authorized_keys
ssh-rsa AAAAB3NzaC1yc2... Juju:juju-client-key
ssh-ed25519 AAAAC3NzaC... Juju:nicolas@thinkpad
ssh-rsa AAAAB3NzaC1yc2... Juju:juju-system-key
```

### Test 2
We also want to make sure that the system ssh key is not settable at bootstrap time.
Running: `juju bootstrap --config system-ssh-keys="..."` should result in ERROR "system-ssh-keys" is not a user configurable item. Please unset this and continue.

## Documentation changes

N/A

## Links

**Jira card:** JUJU-5094

### CODE REVIEW

File: apiserver/facades/agent/keyupdater/authorisedkeys.go [hunk 1]
```diff
@@ -27,11 +34,12 @@ type KeyUpdater interface {
...
-	getCanRead common.GetAuthFunc
+	controllerConfigService ControllerConfigService
+	state                   *state.State
+	model                   *state.Model
+	resources               facade.Resources
+	authorizer              facade.Authorizer
+	getCanRead              common.GetAuthFunc
 }
 
 var _ KeyUpdater = (*KeyUpdaterAPI)(nil)
```
Comment (nvinuesa) [inline apiserver/facades/agent/keyupdater/authorisedkeys.go:45]: The comments in the methods `func (api *KeyUpdaterAPI) WatchAuthorisedKeys()` and `func (api *KeyUpdaterAPI) AuthorisedKeys()` should be changed: `... stored in the model and controller config.`

**Review by nvinuesa — COMMENTED** [review]
Small comment, I can't QA yet though it doesn't build

**Comment by jack-w-shaw** [comment]
`ControllerSuite.TestControllerAndModelConfigInitialisation` in `github.com/juju/juju/state` is failing. Looking like a legit failure. Something probably just needs a bump

**Comment by jack-w-shaw** [comment]
`juju bootstrap --config system-ssh-keys="..."` doesn't seem to do anything

Is this something we want?

**Comment by tlm** [comment]
Thanks @jack-w-shaw,

That was a good pickup. I have updated the PR so that bootstrap will fail if this is set. This isn't part of what is user changeable through the API so we should never allow setting it.

**Comment by jack-w-shaw** [comment]
@tlm This is better. But:
```
$ juju bootstrap -h
...
Available keys for use with --config are:
...
    system-ssh-keys:
      type: string
      description: Defines the system ssh keys
```

It seems our config code isn't really designed for entirely read-only config items. A bit of work is probably required within our config code to properly support this new type of config item. I think this would be better than special casing this one key in cmd

**Comment by tlm** [comment]
@jack-w-shaw we won't be dealing with this in this PR. I'll add it to our tech debt backlog as part of 4.0 milestone to come back to. At the moment I need the original fix undone so it unblocks modelconfig work that has the higher priority.

I would love to fix every part of this up but at the moment this is about undoing the previous work so we don't special case the controller model anymore, so I can get some other work unblocked.


If anyone else feels differently about this I am happy to do the work but advice at the moment is move it along.

**Comment by jack-w-shaw** [comment]
Cool, so long as it's being tracked

**Review by jack-w-shaw — APPROVED** [review]

File: apiserver/facades/agent/keyupdater/authorisedkeys.go [hunk 2]
```diff
@@ -13,11 +14,17 @@ import (
 	"github.com/juju/juju/apiserver/common"
 	apiservererrors "github.com/juju/juju/apiserver/errors"
 	"github.com/juju/juju/apiserver/facade"
+	"github.com/juju/juju/controller"
 	"github.com/juju/juju/rpc/params"
 	"github.com/juju/juju/state"
 	"github.com/juju/juju/state/watcher"
 )
 
+// ControllerConfigService is the interface that gets ControllerConfig form DB.
```
Comment (manadart) [inline apiserver/facades/agent/keyupdater/authorisedkeys.go:23]: "_from_ DB"?

**Pull request merged**